io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/internal/simpletype/SimpleTypeUtil.java

```diff
@@ -119,6 +134,19 @@ else if (simpleTypeValue instanceof XmlDateTime) {
 			}
 		}
 
+		if (value instanceof BigDecimal) {
+			// Create a DecimalFormat object without scientific notation
+			DecimalFormat decimalFormat = new DecimalFormat("0.##############");
+			// Format the BigDecimal
+			return decimalFormat.format(value);
+		}
+		else
+		// Handle primitive types
+		if (value instanceof Double || value instanceof Float || value instanceof Long
+				|| value instanceof Integer) {
+			return String.valueOf(value);
+		}
```

Comment:
If the XML element/attribute has a number type, shouldn't it be handled by an xmlbeans type conversion before? I think if we still end up here, we should rather add the missing conversion.

Also this representation would basically be what was used originally, which can result in scientific notation.